app/s3.py

```diff
@@ -0,0 +1,51 @@
+"""Utilities for working with S3.
+
+In LSST the Docs, ltd-mason is responsible for uploading documentation
+resources to S3. ltd-keeper only really needs to delete resources.
+"""
+
+import logging
+from pprint import pformat
+import boto3
+
+from .exceptions import S3Error
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+def delete_directory(bucket_name, root_path):
+    """Delete all objects in the S3 bucket named `bucket_name` that are
+    found in the `root_path` directory.
+
+    Parameters
+    ----------
+    bucket_name : str
+        Name of an S3 bucket.
+    root_path : str
+        Directory in the S3 bucket that will be deleted.
+    """
+    s3 = boto3.resource('s3')
+    bucket = s3.Bucket(bucket_name)
+
+    if not root_path.endswith('/'):
+        root_path += '/'
+
+    delete_keys = {'Objects': []}
+    key_objects = [{'Key': obj.key}
+                   for obj in bucket.objects.filter(Prefix=root_path)]
+    assert len(key_objects) > 0
```

Comment:
This is to catch the `root_path` not existing in the bucket?  If this is expected to be a common occurrence, you may want to consider adding a more descriptive exception message.